src/components/Checkbox/index.tsx

```diff
@@ -181,107 +183,113 @@ type CheckboxProps = Omit<ReakitCheckboxProps, 'checked'> & {
 } & Required<Pick<ReakitCheckboxProps, 'onChange'>>
 
-const Checkbox = ({
-  checked = false,
-  onChange,
-  onFocus,
-  onBlur,
-  error,
-  name,
-  value,
-  size = 24,
-  children,
-  progress = false,
-  disabled = false,
-  autoFocus = false,
-  className,
-  'data-visibility': dataVisibility,
-}: CheckboxProps) => {
-  const hasChildren = !!children
-  const checkbox = useCheckboxState({ state: checked })
-  const { state, setState } = checkbox
-
-  const computedName = useMemo(() => {
-    if (!name) return getUUID('checkbox')
-
-    return name
-  }, [name])
-
-  const icon = useMemo(() => {
-    if (state === 'indeterminate') return 'minus-box-outline'
-    if (state) return 'checkbox-marked-outline'
-
-    return 'checkbox-blank-outline'
-  }, [state])
-
-  useEffect(() => {
-    setState(checked)
-  }, [checked, setState])
-
-  const onLocalChange = useCallback(
-    (event: ChangeEvent<HTMLInputElement>) => {
-      if (!progress) onChange(event)
-      setState(event.target.checked)
-    },
-    [onChange, progress, setState],
-  )
-
-  const onKeyDown: KeyboardEventHandler = useCallback(
-    event => {
-      if (event.key.charCodeAt(0) === 32) {
-        onChange(event)
-      }
-    },
-    [onChange],
-  )
-
-  return (
-    <>
-      {progress ? (
-        <StyledActivityContainer hasChildren={hasChildren}>
-          <Loader active size={size} />
-        </StyledActivityContainer>
-      ) : null}
-      <StyledCheckBoxContainer
-        className={className}
-        aria-disabled={disabled}
-        data-visibility={dataVisibility}
-      >
-        <StyledReakitCheckbox
-          aria-invalid={!!error}
-          aria-describedby={error ? `${computedName}-hint` : undefined}
-          aria-checked={
-            state === 'indeterminate' ? 'mixed' : (state as boolean)
-          }
-          checked={state === 'indeterminate' ? false : (state as boolean)}
-          size={size}
-          onChange={onLocalChange}
-          onFocus={onFocus}
-          onBlur={onBlur}
-          onKeyDown={onKeyDown}
-          disabled={disabled}
-          value={value}
-          name={computedName}
-          autoFocus={autoFocus}
-        />
-        {!progress ? (
-          <StyledIcon name={icon} size={size} viewBox="0 0 24 24">
-            {state === 'indeterminate' ? (
-              <CheckboxMixedIcon />
-            ) : (
-              <CheckboxIcon />
-            )}
-          </StyledIcon>
+const Checkbox = forwardRef(
+  (
+    {
+      checked = false,
+      onChange,
+      onFocus,
+      onBlur,
+      error,
+      name,
+      value,
+      size = 24,
+      children,
+      progress = false,
+      disabled = false,
+      autoFocus = false,
+      className,
+      'data-visibility': dataVisibility,
+    }: CheckboxProps,
+    ref: ForwardedRef<HTMLLabelElement>,
+  ) => {
+    const hasChildren = !!children
+    const checkbox = useCheckboxState({ state: checked })
+    const { state, setState } = checkbox
```

Comment:
I don't see any other usage of `checkbox`:

```suggestion
    const { state, setState } = useCheckboxState({ state: checked })
```